backend/infrastructures/apps/load/admin.py

```diff
@@ -1,12 +1,102 @@
+from datetime import datetime
+
+from django import forms
 from django.contrib import admin
 from django.db import models
+from django.utils.translation import gettext_lazy as _
+from import_export import admin as import_export_admin
+from import_export import fields, resources
+from import_export.forms import ExportForm
 
 from .models import Data
 
 
-@admin.register(Data)
-class OutputDataAdmin(admin.ModelAdmin):
-    list_display = ("id", "full_name", "age", "is_satisfied")
+class CustomExportForm(ExportForm):
+    timestamp_from = forms.DateTimeField(label=_("Timestamp from"), required=False)
+    timestamp_to = forms.DateTimeField(label=_("Timestamp to"), required=False)
+
+
+class DataResource(resources.ModelResource):
+    def __init__(self, **kwargs):
+        super().__init__()
+        self.timestamp_from = kwargs.get("timestamp_from")
+        self.timestamp_to = kwargs.get("timestamp_to")
+
+    full_name = fields.Field(column_name="full_name")
+    is_satisfied = fields.Field(column_name="is_satisfied")
+    age = fields.Field(column_name="age")
+
+    class Meta:
+        model = Data
+        fields = ("id", "full_name", "is_satisfied", "age", "created_at")
+        export_order = (
+            "id",
+            "full_name",
+            "is_satisfied",
+            "age",
+            "created_at",
+        )
+
+    def dehydrate_full_name(self, obj):
+        return obj.data["full_name"]
+
+    def dehydrate_is_satisfied(self, obj):
+        return obj.data["is_satisfied"]
+
+    def dehydrate_age(self, obj):
+        return obj.data["age"]
+
+    def filter_export(self, queryset, **kwargs):
+        try:
+            if self.timestamp_from:
+                queryset = queryset.filter(created_at__gte=self.timestamp_from)
+        except ValueError:
+            pass
+
+        try:
+            if self.timestamp_to:
+                queryset = queryset.filter(created_at__lt=self.timestamp_to)
+        except ValueError:
+            pass
+
+        return queryset
+
+
+class DataAdminForm(forms.ModelForm):
+    class Meta:
+        model = Data
+        fields = "__all__"
+
+    def clean_data(self):
+        data = self.cleaned_data.get("data")
+        if not isinstance(data, dict):
+            raise forms.ValidationError(_("Invalid JSON data."))
+
+        required_fields = {
+            "full_name": str,
+            "age": int,
+            "is_satisfied": bool,
+        }
+
+        for field, expected_type in required_fields.items():
+            if field not in data:
+                raise forms.ValidationError(_(f"Missing field '{field}' in JSON data."))
+            if not isinstance(data[field], expected_type):
```

Comment:
According to me this check is redundant. Assumtion is, that data are always correct. We read them. 


Did i miss something?

Reply:
it's needed, so admin always passes correct data type , for example age must be int (11) and not string ("11")